docker/Dockerfile-dev

```diff
@@ -5,127 +5,73 @@
 
 FROM golang:1.16-buster
 
-# Avoid warnings by switching to noninteractive
-ENV DEBIAN_FRONTEND=noninteractive
+# Optionally install zsh shell.
+ARG INSTALL_ZSH="true"
```

Comment:
Support bash is enough I think, we can introduce zsh if more need from users

Reply:
Yeah, I don't feel strongly about this; it was just kept as the default setting that GitHub Codespaces use for their image. Any other folks want to weigh in? Given the support for dotfiles, users who want ZSH could add it as part of their install scripts.